### PR TITLE
3 view list of comments associated with article

### PR DESCRIPTION
**Pull request by AliCW**

### CODE REVIEW

File: src/components/SingleArticle.jsx [hunk 1]
```diff
@@ -6,32 +6,72 @@ import * as api from '../api'
...
         })
     }, [article_id]
     )
 
+    const upVote = (article_id) => {
+        if(addVotes === false) {
+            setAddVotes(true)
+                selectArticle((article) => {
+                    return article.map((votes) => { 
+                        console.log(article)
```
Comment (popatre) [inline src/components/SingleArticle.jsx]: remove log

File: src/components/SingleArticle.jsx [hunk 2]
```diff
@@ -6,32 +6,72 @@ import * as api from '../api'
...
+            downVote(article[0].article_id)
+        }
+    }
+  
+        const downVote = (article_id) => {
+            if(subVotes === false) {
+                setSubVotes(true)
+                    selectArticle((article) => {
+                        return article.map((votes) => { 
+                            console.log(article)
```
Comment (popatre) [inline src/components/SingleArticle.jsx]: log

File: src/components/SingleArticle.jsx [hunk 3]
```diff
@@ -6,32 +6,72 @@ import * as api from '../api'
...
+                            return {...votes, votes: article[0].votes - 1}
+                        })
+                    })
+                    api.decVote(article_id)
```
Comment (popatre) [inline src/components/SingleArticle.jsx]: Needs some error handling for when the api request fails - at the moment, if looks like its gone well until you refresh the page

File: src/components/SingleArticle.jsx [hunk 4]
```diff
@@ -6,32 +6,72 @@ import * as api from '../api'
...
+    const upVote = (article_id) => {
+        if(addVotes === false) {
+            setAddVotes(true)
+                selectArticle((article) => {
+                    return article.map((votes) => { 
+                        console.log(article)
+                        return {...votes, votes: article[0].votes + 1}
+                    })
+            })  
+            api.incVote(article_id)
```
Comment (popatre) [inline src/components/SingleArticle.jsx]: Needs some error handling for when the api request fails - at the moment, if looks like its gone well until you refresh the page

**Review by popatre — CHANGES_REQUESTED** [review]
Seems to all work well, just needs the error handling added, please

File: src/components/SingleArticle.jsx [hunk 5]
```diff
@@ -6,32 +6,82 @@ import * as api from '../api'
...
+            setAddVotes(true)
+            selectArticle((article) => {
+                return article.map((votes) => {
+                    return { ...votes, votes: article[0].votes + 1 }
+                })
+            })
+            api.incVote(article_id).catch(() => {
+                selectArticle((article) => {
+                    return article.map((votes) => {
+                        return { ...votes, votes: article[0].votes - 1 }
```
Comment (popatre) [inline src/components/SingleArticle.jsx]: Could you just add one small thing and just have a little message appear to say that something has gone wrong, and to try again?
Just for a better user experience

**Review by popatre — CHANGES_REQUESTED** [review]
Could you just add one small thing and just have a little message appear to say that something has gone wrong, and to try again?
Just for a better user experience

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Ali, looking at your repo, you don't seem to have a separate branch for this ticket. I'm using branch 4 to test your app as it includes the voting functionality. 

- When clicking twice on a vote in the same direction, first the votes increase, then decrease, which shouldn't happen. 
- In case of multiple clicks (more than two) the site breaks and throws a RangeError (Maximum call stack size exceeded).

Please debug these and send us another PR with it.

**Review by popatre — APPROVED** [review]
Presume this is id ticket 7 - if it is - all good

**Pull request merged**